src/Data/List/Relation/Binary/Permutation/Propositional/Properties.agda

```diff
@@ -28,9 +30,31 @@ open import Relation.Unary using (Pred)
 open import Relation.Binary
 open import Relation.Binary.PropositionalEquality as ≡
   using (_≡_ ; refl ; cong; cong₂; _≢_; inspect)
+open import Relation.Nullary
 
 open PermutationReasoning
 
+
+------------------------------------------------------------------------
+-- Permutations of empty and singleton lists
+
+module _ {a} {A : Set a} where
+
+  ↭-[]-inv : ∀ {xs : List A} → xs ↭ [] → xs ≡ []
+  ↭-[]-inv refl = refl
+  ↭-[]-inv (trans p q) with ↭-[]-inv q
+  ... | refl with ↭-[]-inv p
+  ... | refl = refl
+
+  ¬∷↭[] : ∀ {x} {xs : List A} → ¬ ((x ∷ xs) ↭ [])
```

Comment:
I think it's more standard to call this `¬x∷xs↭[]`, e.g. https://github.com/agda/agda-stdlib/blob/b859bd363a96bc862ead0509bdf5869837651896/src/Data/Nat/Properties.agda#L85 and